compiler/luci/service/src/Nodes/CircleDiv.test.cpp

```diff
@@ -44,3 +46,262 @@ TEST(CloneNodeTest, clone_Div_NEG)
+  div.y(&input_2);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&div, shape));
+  ASSERT_EQ(4, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(4, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(1, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, div_dynamic_shape_non_1)
+{
+  luci::CircleInput input_1;
+  luci::CircleInput input_2;
+  luci::CircleDiv div;
+
+  input_1.shape({1, 4, 3, 1});
+  input_1.shape_status(luci::ShapeStatus::VALID);
+
+  input_2.shape({1, 1, 1});
+  input_2.shape_status(luci::ShapeStatus::VALID);
+  input_2.dim(0).unset();
+
+  div.x(&input_1);
+  div.y(&input_2);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&div, shape));
+  ASSERT_EQ(4, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(4, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(1, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, div_dynamic_shape_1)
+{
+  luci::CircleInput input_1;
+  luci::CircleInput input_2;
+  luci::CircleDiv div;
+
+  input_1.shape({1, 4, 3, 1});
+  input_1.shape_status(luci::ShapeStatus::VALID);
+
+  input_2.shape({1, 1, 1});
+  input_2.shape_status(luci::ShapeStatus::VALID);
+  input_2.dim(2).unset();
+
+  div.x(&input_1);
+  div.y(&input_2);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&div, shape));
+  ASSERT_EQ(4, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_FALSE(shape.dim(3).known());
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(4, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(0, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, div_dynamic_shape_both)
+{
+  luci::CircleInput input_1;
+  luci::CircleInput input_2;
+  luci::CircleDiv div;
+
+  input_1.shape({1, 4, 3, 1});
+  input_1.shape_status(luci::ShapeStatus::VALID);
+  input_1.dim(3).unset();
+
+  input_2.shape({1, 1, 1});
+  input_2.shape_status(luci::ShapeStatus::VALID);
+  input_2.dim(2).unset();
+
+  div.x(&input_1);
+  div.y(&input_2);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&div, shape));
+  ASSERT_EQ(4, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_FALSE(shape.dim(3).known());
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(4, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(0, shape.dim(3).value());
```

Comment:
In case of `unknown shape`, the value of dim is alway 0, right?
Conversely, if the value of dim is 0, should it be always `unknown shape`?
I don't think so, but I'll leave it here for clarity =)

Reply:
There is `ASSERT_FALSE(shape.dim(3).known());` that checks it's unknown.
I think @shs-park has a point that we can give an incorrect idea that unknown has value of 0, which is not intended.

```suggestion
```

Reply:
I used `unset` method of `Dimension` to make a dimension unknown. `unset` makes `known()` return false, and sets the value 0. I think the value 0 is not that important. `ASSERT_FALSE(shape.dim(3).known());` is important as @seanshpark  mentioned. But it's aligned with implementation of `unset`.